### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Somdotta07**
- Created class `Person` 
   - Instance vars: `@id`, `@name`, and `@age`.
   - Constructor with `name`, `age`, and `parent_permission` as parameter.` name` and `parent_permission` are optional and have default values of "`Unknown`" and `true`
   - Used setters and getters and set Private `is_of_age?` and Public method `can_use_services?`
-  Created class `Student` 
   - Inherited from `Person.`
   - The constructor extends the parent's constructor by adding `@classroom` and a parameter for it.
   - Method `play_hooky` that returns` "¯\(ツ)/¯".`
-  Created class `Teacher` 
   - Inherited from `Person.`
   - The constructor extends the parent's constructor by adding `@specialization` and a parameter for it.
   - Override `can_use_services?` so it always returns` true`.
- Each class is saved in a separate file

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+
+class Teacher < Person
+  def initialize(specialization, _classrom, age, name = 'Unknown', parent_permission: true)
```
Comment (mikethreels) [inline teacher.rb]: - [x] classroom is not a argument for the Teacher class, therefor it is not needed to add it to the initialize method.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative './person'
+
+class Student < Person
+  def initialize(_classrom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (mikethreels) [inline student.rb]: - [x] you're creating the instance vairable `@classroom` using `classroom` however the name of the argument you're passing is `_classrom` please make sure fix this.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Somdotta, 

Good job creating the `Person class` and including the private method.
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 2 above)
Comment (Somdotta07) [reply student.rb]: Thank you for your review, I'm extremely sorry for my mistake, I'll fix it ASAP

File: teacher.rb (hunk 1 above)
Comment (Somdotta07) [reply teacher.rb]: Thanks for detecting the error, will fix it.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Somdotta07,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**